Gemfile

```diff
@@ -42,14 +42,14 @@ supports_windows = true
 
 group :development do
   gem 'puppet-lint',                        :require => false
-  gem 'metadata-json-lint',                 :require => false if ! supports_windows || Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.3.1')
+  gem 'metadata-json-lint',                 :require => false, :platforms => 'ruby'
```

Comment:
@hunner @Iristyle Bundler is weird. When you are seeing it do all of the resolving in DEBUG, that doesn't mean those things are still installed. So while you see it resolve json, if those gems are excluded with (`--without`), it won't install them or their dependencies.